internal/locate/region_request.go

```diff
@@ -2211,7 +2222,7 @@ func (s *RegionRequestSender) onSendFail(bo *retry.Backoffer, ctx *RPCContext, r
 			}
 		}
 	}
-	metrics.TiKVRPCErrorCounter.WithLabelValues(errors.Cause(err).Error(), storeLabel).Inc()
+	metrics.TiKVRPCErrorCounter.WithLabelValues(getErrMsg(err), storeLabel).Inc()
```

Comment:
Should it be like this?
```
errStr := getErrMsg(err)
if len(errStr) > 0 {
    metrics.TiKVRPCErrorCounter.WithLabelValues(getErrMsg(err), storeLabel).Inc()
}
```
Or to check if the input `err` is nil?
It's strange to use an empty string as a metric label.

Reply:
But normally, the `errStr` shouldn't be an empty string, since the `err` can't be nil here.

Reply:
If the input `err` is required not to be nil, then should we return directly at L2173, or add some assertions?

But how the panic in https://github.com/tikv/client-go/issues/1343 happen? Is the input `err` nil in that case?

Reply:
#1343 is caused by some error implement `Cause` interface, but `Cause` method return `nil`. See the test in this PR.

Reply:
So the 
```
if len(errStr) > 0 { metrics.TiKVRPCErrorCounter.WithLabelValues(getErrMsg(err), storeLabel).Inc() }
```
seems to be necessary, as the cause string could be empty and it should not be used as label.

Or we use `unknown` for this situation.

Reply:
Skipping recording error counting if the `cause` is empty, may lead to incorrect total error number counting.
Perhaps `Unknown` is better?

Reply:
In `getErrMsg`, if the error's cause is nil, will return the `error.Error()`, unless the error message itself is empty string, otherwise it wouldn't be empty string here.

Reply:
It's better to use some defensive programming style but not to assume if it could be nil or not.
Actually if it's c/c++ code, each pointer should be checked first before using.

Reply:
done.